Documentation/contributing/development/dev_setup.rst

```diff
@@ -280,6 +288,21 @@ with NFS. Note that your host firewall must have a variety of ports open. The
 Vagrantfile will inform you of the configuration of these addresses and ports
 to enable NFS.
 
+.. note::
+
+   Firewalld users must add the virtualbox interface to their default
+   trusted zone and allow the nfs, rpc-bind, and mountd services.
+
+   For example on Fedora >= 33:
+
+   .. code-block:: none
+  
+   # sudo firewall-cmd --permanent --add-interface=vboxnet0 --zone=FedoraWorkstation
+   # sudo firewall-cmd --add-service=nfs      --permanent
+   # sudo firewall-cmd --add-service=rpc-bind --permanent
+   # sudo firewall-cmd --add-service=mountd   --permanent
+   # sudo firewall-cmd --reload
+
```

Comment:
I'm fine with having this but `vboxnet0` might not be the interface used for the VM. Also we have `contrib/vagrant/nftables.rules` where devs can run `sudo nft -f contrib/vagrant/nftables.rules` to avoid hitting the firewall issues.

Reply:
Yeah, this section maybe too specific. 

I did however, miss the nftables.rules - maybe I rip this out and just make a note that the nft rules exists and should be ran before `vagrant up` ?

Reply:
Yeah, this is usually what I run `make build && sudo nft -f contrib/vagrant/nftables.rules && NO_BUILD=1 NETNEXT=1 NWORKERS=1 NFS=1 K8S=1 contrib/vagrant/start.sh`

Reply:
agreed, this is better then leaking fedora specific instructions in there. I'll replace.

Reply:
I run into issues with running the nfttables.rules - I think because the necessary chain does not exist by default in the firewalld table:

```
~/git/go/cilium louis/docs-updates
❯ cat contrib/vagrant/nftables.rules
insert rule inet firewalld filter_IN_public_allow ip saddr 192.168.34.0/24 tcp dport 20048 ct state { 0x8, 0x40 } accept
insert rule inet firewalld filter_IN_public_allow ip saddr 192.168.34.0/24 tcp dport 2049 ct state { 0x8, 0x40 } accept
insert rule inet firewalld filter_IN_public_allow ip saddr 192.168.34.0/24 tcp dport 111 ct state { 0x8, 0x40 } accept

~/git/go/cilium louis/docs-updates
❯ sudo nft -f contrib/vagrant/nftables.rules 
contrib/vagrant/nftables.rules:1:28-49: Error: No such file or directory; did you mean chain ‘filter_IN_docker_allow’ in table inet ‘firewalld’?
insert rule inet firewalld filter_IN_public_allow ip saddr 192.168.34.0/24 tcp dport 20048 ct state { 0x8, 0x40 } accept
                           ^^^^^^^^^^^^^^^^^^^^^^
contrib/vagrant/nftables.rules:2:28-49: Error: No such file or directory; did you mean chain ‘filter_IN_docker_allow’ in table inet ‘firewalld’?
insert rule inet firewalld filter_IN_public_allow ip saddr 192.168.34.0/24 tcp dport 2049 ct state { 0x8, 0x40 } accept
                           ^^^^^^^^^^^^^^^^^^^^^^
contrib/vagrant/nftables.rules:3:28-49: Error: No such file or directory; did you mean chain ‘filter_IN_docker_allow’ in table inet ‘firewalld’?
insert rule inet firewalld filter_IN_public_allow ip saddr 192.168.34.0/24 tcp dport 111 ct state { 0x8, 0x40 } accept
```

I think this script could create the necessary chain in the firewalld table before adding the rules and all would work across any firewalld controlled distribution. 

I checked firewalld's docs and the software consistently uses the "firewalld" table. 

What do you think about adding the chain creation to this script?

Reply:
It makes sense to me that if `nftables.rules` is going to add rules to that table, it should also attempt to create the table if it doesn't exist. :+1:

Reply:
Been playing around with my fedora machine and nfttables, the way firewalld configures nftables makes it difficult to have a single "nfttables.rules" file that will work across the board. Firewalld is doing a lot of jumping to other Fedora specific chains like: 

```
	chain filter_INPUT_ZONES {
		iifname "br-b4f835e4fdb9" goto filter_IN_docker
		iifname "docker0" goto filter_IN_docker
		iifname "wlp0s20f3" goto filter_IN_FedoraWorkstation
		iifname "virbr0" goto filter_IN_FedoraWorkstation
		goto filter_IN_FedoraWorkstation
	}

```
I'm going to rip this out for now, and think about this a bit more in a different PR.